### PR TITLE
feat: Manual sign up

### PR DESCRIPTION
**Pull request by lucamouchel**
This PR adds the possibility to sign up using email. However there is no email verification as that would be a bit tedious to implement.

()

### CODE REVIEW

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java [hunk 1]
```diff
@@ -20,35 +29,135 @@ public class SignUpActivity extends AppCompatActivity {
...
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+            issues.setText("");
+            if (email.getText().toString().matches(EMAIL_REGEX)) {
+                setEnabled(s.toString().matches(PW_REGEX));
+            } else {
```
Comment (hmiranda-queiros) [inline app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:47]: Maybe add a comment to explain what happens if password and mail are correct, it is not clear when we read it like that

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java [hunk 2]
```diff
@@ -20,35 +29,135 @@ public class SignUpActivity extends AppCompatActivity {
...
+                else if (!exists.get()) {
+                    issues.setText("Wrong sign in credentials");
+                } else {
+                    Authenticator.signIn(this);
+                }
+                return null;
+            });
+        } else {
+            password.setClickable(false);
+        }
```
Comment (hmiranda-queiros) [inline app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:131]: What does this line mean ? If the password format is not correct we can't click on it ?

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java [hunk 3]
```diff
@@ -20,35 +29,135 @@ public class SignUpActivity extends AppCompatActivity {
...
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
+
+    private void setEnabled(boolean enabled) {
+        float alpha = enabled ? 1f : 0.3f;
+        signUp.setAlpha(alpha);
+        signIn.setAlpha(alpha);
+        signIn.setClickable(enabled);
```
Comment (hmiranda-queiros) [inline app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:148]: Why only SignIn button is set clickable and not the SignUp one too ?

**Review by hmiranda-queiros — COMMENTED** [review]
Great Work ! This feature will be useful ! I just put some comments to ask some details about your implementation. Maybe also add a screenshot of the UI if you can please.

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java (hunk 3 above)
Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:148]: this way if a user clicks on signup they get a message to tell them why their password or email is not valid

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java (hunk 2 above)
Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:131]: good point idk

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java [hunk 4]
```diff
@@ -20,35 +29,135 @@ public class SignUpActivity extends AppCompatActivity {
...
+        }
+        if (pw.length() < 8) {
+            issues.setText("Password should be minimum 8 characters");
+            return;
+        }
+
+        if (pw.matches(PW_REGEX)) {
+            Authenticator.manualSignUp(email.getText().toString(), password.getText().toString()).thenCompose((success) -> {
+                if (!success.get()) {
+                    issues.setText("This email is already used");
```
[the remark this answers is not in the document]
Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:103]: no it is if a user already has an account registered to that email and they try to create another account with the same email

[the remark this answers is not in the document]
Comment (lucamouchel) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:104]: but yes it also works like that in the case u described

[the remark this answers is not in the document]
Comment (hmiranda-queiros) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:104]: ok great !

File: app/src/main/java/ch/epfl/culturequest/SignUpActivity.java (hunk 3 above)
Comment (hmiranda-queiros) [reply app/src/main/java/ch/epfl/culturequest/SignUpActivity.java:149]: oh ok

**Review by hmiranda-queiros — APPROVED** [review]
All good to go !

**Pull request merged**